streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractSegments.java

```diff
@@ -117,12 +118,20 @@ public void openExisting(final ProcessorContext context, final long streamTime)
     }
 
     @Override
-    public List<S> segments(final long timeFrom, final long timeTo) {
+    public List<S> segments(final long timeFrom, final long timeTo, final boolean forward) {
         final List<S> result = new ArrayList<>();
-        final NavigableMap<Long, S> segmentsInRange = segments.subMap(
-            segmentId(timeFrom), true,
-            segmentId(timeTo), true
-        );
+        final NavigableMap<Long, S> segmentsInRange;
+        if (forward) {
```

Comment:
nit: we can just call `subMap` out of the condition and only call `descendingMap()` based on the condition.

Reply:
I'd prefer this immutable approach as it's the same used in keyValueStore changes, if that's ok.

Reply:
SG, `nit` comments are just suggestions not mandatory.